runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsOverlay.java

```diff
@@ -0,0 +1,294 @@
+			int foodHealValue = 0;
+			final MenuEntry[] menu = client.getMenuEntries();
+			final int menuSize = menu.length;
+			final MenuEntry entry = menu[menuSize - 1];
+			final Effect change = itemStatChanges.get(entry.getIdentifier());
+
+			if (change != null & entry.getParam1() == WidgetInfo.INVENTORY.getId())
+			{
+				final StatsChanges statsChanges = change.calculate(client);
+				for (final StatChange c : statsChanges.getStatChanges())
+				{
+					if (c.getStat().getName().equals("Hitpoints"))
+					{
+						foodHealValue = Integer.parseInt(c.getTheoretical());
+					}
+
+					if (c.getStat().getName().equals("Prayer"))
+					{
+						prayerHealValue = Integer.parseInt(c.getTheoretical());
+					}
+				}
+			}
+
+			renderHealingBar(g, offsetHealthX, offsetHealthY,
+					maxHealth, currentHealth, height,
+					foodHealValue, HEAL_COLOR);
+
+			renderHealingBar(g, offsetPrayerX, offsetPrayerY,
+					maxPrayer, currentPrayer, height,
+					prayerHealValue, PRAYER_HEAL_COLOR);
+		}
+
+		if (config.enableSkillIcon() || config.enableCounter())
+		{
+			renderIconsAndCounters(g, offsetPrayerX, offsetPrayerY, prayerImage.getScaledInstance(IMAGE_SIZE, IMAGE_SIZE, Image.SCALE_SMOOTH), counterPrayerText, PRAYER_LOCATION_X);
+			renderIconsAndCounters(g, offsetHealthX, offsetHealthY, healthImage, counterHealthText, HEALTH_LOCATION_X);
+		}
+
+		return null;
+	}
+
+	private void renderBar (Graphics2D graphics, int x, int y, int max, int current, int height, Color filled)
+	{
+		graphics.setColor(BACKGROUND);
+		graphics.drawRect(x, y, WIDTH - PADDING, height - PADDING);
+		graphics.fillRect(x, y, WIDTH, height);
+
+		int filledHeight = getBarHeight(max, current, height);
+		graphics.setColor(filled);
+		graphics.fillRect(x + PADDING,
+				y + PADDING + (height - filledHeight),
+				WIDTH - PADDING * OFFSET,
+				filledHeight - PADDING * OFFSET);
+	}
+
+	private void renderHealingBar(Graphics2D graphics, int x, int y, int max, int current, int height, int heal, Color color)
+	{
+		int filledHeight = getBarHeight(max, heal, height);
+		int filledCurrentHeight = getBarHeight(max, current, height);
+		int overHeal;
+		graphics.setColor(color);
+
+		if (filledHeight + filledCurrentHeight > height)
+		{
+			overHeal = filledHeight + filledCurrentHeight - height;
+			filledHeight = filledHeight - overHeal + OVERHEAL_OFFSET;
+			graphics.setColor(OVERHEAL_COLOR);
+			graphics.fillRect(x + PADDING,
+					y - filledCurrentHeight + (height - filledHeight) + HEAL_OFFSET,
+					WIDTH - PADDING * OVERHEAL_OFFSET,
+					filledHeight - PADDING * OVERHEAL_OFFSET);
+		}
+		else
+		{
+			graphics.fillRect(x + PADDING,
+					y - OVERHEAL_OFFSET - filledCurrentHeight + (height - filledHeight) + HEAL_OFFSET,
+					WIDTH - PADDING * OVERHEAL_OFFSET,
+					filledHeight + OVERHEAL_OFFSET - PADDING * OVERHEAL_OFFSET);
+		}
+	}
+
+	private void renderIconsAndCounters(Graphics2D graphics, int x, int y, Image image, String counterText, int counterPadding)
```

Comment:
This looks like it can be `static`?

```suggestion
	private static void renderIconsAndCounters(Graphics2D graphics, int x, int y, Image image, String counterText, int counterPadding)
```

Reply:
it can't be static because i can't access the config then to make changes on the position of the icon/counters when they change.

Reply:
Missed that one, good point.